Description: Avoid using deprecated ast.Num and node.n
Origin, upstream, https://github.com/joblib/joblib/commit/05caf0772d605799e5d2337018fd32ac829b37aa
Author: Loïc Estève <loic.esteve@ymail.com>
Last-Update: 2023-10-20

--- a/joblib/_utils.py
+++ b/joblib/_utils.py
@@ -35,8 +35,8 @@
 
 
 def eval_(node):
-    if isinstance(node, ast.Num):  # <number>
-        return node.n
+    if isinstance(node, ast.Constant):  # <constant>
+        return node.value
     elif isinstance(node, ast.BinOp):  # <left> <operator> <right>
         return operators[type(node.op)](eval_(node.left), eval_(node.right))
     elif isinstance(node, ast.UnaryOp):  # <operator> <operand> e.g., -1
